### PR TITLE
fix: remove enrichContextUsed fallback producing wrong ContextUsedTokens

### LINKED ISSUE #35: Bug: token counter wrong for non-Claude backends
## Problem

`ContextUsedTokens` is wrong for non-Claude backends. Codex is reporting values like 5861k (5.8 million tokens), which is impossible.

The token counting logic likely only works correctly for Claude's output format. Other backends (Codex, OpenCode) and engine types (ACP) either don't emit the same usage data or it's being parsed/accumulated incorrectly.

Note: context window capacity (`ContextSizeTokens`) is not expected from agentrun — CLIs don't report it and it varies by provider. Foundry hardcodes known sizes on its side. This issue is only about the used-token counter being wrong.

### PR DESCRIPTION
**Pull request by dmora**
## Summary

- **Remove `enrichContextUsed`** from `engine/cli/process.go` — the fallback summed all token fields (including OutputTokens, ThinkingTokens) as a context-fill estimate, producing wildly inflated `ContextUsedTokens` for Codex/OpenCode backends
- **Leave `ContextUsedTokens` unset (0)** when CLI engine lacks per-call usage data, rather than fabricating incorrect values
- **Update docs** (`message.go` godoc, `CLAUDE.md`) to clarify CLI/ACP semantics and the `omitempty` disambiguation pattern (`ContextSizeTokens > 0` = reported but empty)
- **Add JSON regression test** for the "reported but empty" ACP scenario

Fixes #35

## Test plan

- [x] `TestApplyContextFill_NoPerCallUsage` updated: expects 0 instead of 6500
- [x] `TestEnrichContextUsed` removed (tested deleted function)
- [x] `TestUsage_JSON_ContextWindowReportedButEmpty` added: locks down omitempty behavior
- [x] `TestCallContextFill` retained (Layer 1 helper unaffected)
- [x] All other `TestApplyContextFill_*` tests pass unchanged
- [x] `make qa` green (lint, test-race, vet, vulncheck, examples)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ Per-field semantics for `Message` metadata. Refer to godoc for full documentatio
...
+- **ContextUsedTokens on CLI MessageResult**: The CLI engine derives `ContextUsedTokens` from the peak per-call input-side fill (`max(InputTokens + CacheReadTokens + CacheWriteTokens)`) across non-result messages in the turn. Only set when the backend emits per-call usage (e.g., Claude CLI assistant events). Backends that report usage only on the result event (Codex, OpenCode) leave this field at 0 — the engine intentionally does not fabricate a value from result-level totals. ACP's `usage_update` provides authoritative context fill and is unaffected. Note: `ContextUsedTokens` uses `omitempty`, so 0 is always omitted in JSON. On ACP `MessageContextWindow` where 0 means "empty context window" (fresh session), consumers distinguish "reported but empty" from "not reported" by checking `ContextSizeTokens > 0` on the same message.
```
[the remark this answers is not in the document]
Comment (dmora) [reply CLAUDE.md]: Thanks for the readability suggestion! The existing CLAUDE.md Message Field Reference bullets all follow this same single-paragraph style (e.g., Usage omitempty semantics, StopReason carry-forward, Denials). Breaking this one into a nested list would be inconsistent with the rest of the section. Keeping as-is for style consistency.
